src/Microsoft.TemplateEngine.Cli/New3Command.cs

```diff
@@ -529,16 +537,26 @@ private CreationResultStatus EnterMaintenanceFlow()
                 argsError = !ValidateRemainingParameters();
             }
 
-            if (argsError || IsHelpFlagSpecified)
+            if(IsHelpFlagSpecified)
             {
-                if(commandParseFailureMessage != null)
+                if (commandParseFailureMessage != null)
                 {
                     Reporter.Error.WriteLine(commandParseFailureMessage.Bold().Red());
                 }
 
                 ShowUsageHelp();
                 ShowTemplateHelp(UnambiguousTemplateToUse.Info);
-                return argsError ? CreationResultStatus.InvalidParamValues : CreationResultStatus.Success;
+                return CreationResultStatus.Success;
```

Comment:
Couldn't we also he in a state where we had arg errors? Do we want to return success in that case? Does this translate to the error level for the process?